v2/writer.go

```diff
@@ -0,0 +1,125 @@
+package car
+
+import (
+	"bytes"
+	"context"
+	"github.com/ipfs/go-cid"
+	format "github.com/ipfs/go-ipld-format"
+	car_v1 "github.com/ipld/go-car"
+	"github.com/willscott/carbs"
+	"io"
+)
+
+type (
+	// Padding represents the number of padding bytes.
+	Padding uint64
```

Comment:
Maybe unexport this; it largely seems like an implementation detail to me. The user should only care about passing you uint64, and then you could have a `writePadding` func that does the logic of writing a certain amount of zero memory.

I also think the option methods would be easier to use with uint64 parameters, as otherwise users might have to explicitly convert to carv2.Padding.

Reply:
There are `WithPadding` methods in the Header that use this type. Do you think we should get those to use some other type if we unexpert this?

Reply:
Yes; those could just take uint64, so I don't see a reason to do anything else.